File: 24_10/21/P17684/solution.py
import string

def solution(msg):
    # 글자수
    N = len(msg)
    # 사전 선언
    msg_dict = {string.ascii_uppercase[i] : i + 1 for i in range(26)}
    
    # 글자 압축
    idx = 0
    dict_num = 27
    answer = []
    while (idx < N):
        end = idx + 2
        while (end <= N and msg[idx:end] in msg_dict): end += 1
        if (end <= N): msg_dict[msg[idx:end]] = dict_num
        
        end -= 1
        answer.append(msg_dict.get(msg[idx:end]))
        
        dict_num += 1
        idx = end
    
    return answer

File: 24_10/21/P17684/solution.js
function solution(msg) {
    // 단어 길이
    const N = msg.length;
    
    // 사전 선언
    const dict = {};
    for (let i = 1; i <= 26; i++) {
        dict[String.fromCharCode(i + 64)] = i;
    }
    
    // 압축 진행
    let idx = 0;
    let dictNum = 27;
    const answer = [];
    while (idx < N) {
        let end = idx + 2;
        while (end <= N && dict[msg.slice(idx, end)]) {
            end++;
        }
        if (end <= N) {
            dict[msg.slice(idx, end)] = dictNum++;
        }
        answer.push(dict[msg.slice(idx, --end)]);
        idx = end;
    }

    
    return answer;
}